src/main/scala/ReleaseExtra.scala

```diff
@@ -225,7 +225,7 @@ object ReleaseStateTransformations {
   }
 
   lazy val publishArtifacts = ReleaseStep(
-    action = publishArtifactsAction,
+    action = runPublishArtifactsAction,
```

Comment:
I just created a small test project and tried it out, the publishing is definitely happening for me:

https://github.com/jroper/sbt-release-test